modules/xfem/src/efa/EFAElement3D.C

```diff
@@ -1054,9 +1077,18 @@ EFAElement3D::createChild(const std::set<EFAElement *> & CrackTipElements,
         EFAPoint p(0.0, 0.0, 0.0);
         p = _local_node_coor[j];
         EFAPoint origin_to_point = p - orig;
+
+        bool use_temp = false;
+        for (unsigned int k = 0; k < normal_v.size(); ++k)
+          if (origin_to_point * normal_v[k] > Xfem::tol)
+          {
+            use_temp = true;
+            break;
+          }
+
         if (_fragments.size() == 1 && !shouldDuplicateForCrackTip(CrackTipElements))
           childElem->setNode(j, _nodes[j]); // inherit parent's node
-        else if (origin_to_point * normal < Xfem::tol)
+        else if (!use_temp)
```

Comment:
Why are we relying these geometric checks to see if the node is in the fragment? We have the list of nodes in the fragment in its data structure. Can't we just check to see if it's in that set of nodes?

Is the purpose of this to deal with higher-order elements? If that's the case, maybe we could do both the geometric check and see if it's in the list of nodes?